ui/Comments/index.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from "react";
+import { View, FlatList, StyleSheet } from "react-native";
+import { Overlay } from "react-native-elements";
+
+import MaximisedComment from "./MaximisedComment";
+import ReplyBox from "./ReplyBox";
+import BackButton from "./BackButton";
+
+type Props = {
+  isVisible: boolean;
+  onBack: (arg0?: Function) => void;
+  comments: Array<IComment>;
+};
+
+const CommentBox = (props: Props) => {
+  let [isVisible, setIsVisible] = useState(props.isVisible);
+
+  const BackHandler = () => {
+    props.onBack(setIsVisible);
+  };
+
```

Comment:
I don't get why `props.onBack` is defined like that - why do you need to pass `setIsVisible` into `props.onBack`?

Reply:
I was thinking that from screen one would may instead of re-rendering  may want to just make comment overlay off by using the setIsVisible but later I realized to make it visible again it has to be re-rendered, and forget to remove it. Will remove it.